syncer/syncer.go

```diff
@@ -242,11 +242,13 @@ func (s *syncer) bulkSyncWithPeer(peerID peer.ID, newBlockCallback func(*types.B
 				return lastReceivedNumber, false, fmt.Errorf("unable to verify block, %w", err)
 			}
 
-			if err := s.blockchain.WriteBlock(block, syncerName); err != nil {
+			fullBlock := &types.FullBlock{Block: block}
+
+			if err := s.blockchain.WriteFullBlock(fullBlock, syncerName); err != nil {
```

Comment:
Could we execute the block here explicitly, return a `FullBlock` and then call `WriteFullBlock`?

Reply:
Ok, changed this a bit, as per our discussion. Check it out: https://github.com/0xPolygon/polygon-edge/pull/1110/commits/51ac590f0b5f1539a72ed8f9f4127c0bbcbcdec7